charactersheet/charactersheet/services/common/account/xmpp_connection_service.js

```diff
@@ -107,14 +109,17 @@ function _XMPPService(config) {
                 console.log('Connected.');
             }
 
+            // Send initial presence.
+            // https://xmpp.org/rfcs/rfc3921.html#presence
+            self.connection.send($pres().tree());
+
             Notifications.xmpp.connected.dispatch();
         } else if (status === Strophe.Status.DISCONNECTED) {
             if (self._shouldLog() && 'console' in window) {
                 console.log('Disconnected.');
             }
             Notifications.xmpp.disconnected.dispatch();
         }
+        // Add more logging...
```

Comment:
Not really. There are some unhandled conditions. Reconnection is 1 off the top of my head.